server/utils/utils.ts

```diff
@@ -157,6 +160,26 @@ const isBankHolidayOrWeekend = (date: Moment, bankHolidays: Holiday[]) => {
 const licenceIsTwoDaysToRelease = (licence: Licence) =>
   moment(licence.conditionalReleaseDate, 'DD/MM/YYYY').diff(moment(), 'days') <= 2
 
+const selectReleaseDate = (nomisRecord: Prisoner) => {
+  let dateString = nomisRecord.conditionalReleaseDate
+
+  if (nomisRecord.confirmedReleaseDate) {
+    dateString = nomisRecord.confirmedReleaseDate
+  }
+
+  if (nomisRecord.conditionalReleaseOverrideDate) {
+    dateString = nomisRecord.conditionalReleaseOverrideDate
+  }
+
+  try {
```

Comment:
might be best to check if date is defined or not - try catch suggests the date is an invalid format which should never be the case.